lib/action/haptic_action.dart

```diff
@@ -17,6 +18,9 @@ class HapticAction extends EnsembleAction {
 
   final HapticTypes type;
 
+  factory HapticAction.fromMap(dynamic inputs) =>
+      HapticAction.fromYaml(payload: Utils.getYamlMap(inputs));
+
   factory HapticAction.fromYaml({Map? payload}) {
     if (payload == null || payload['type'] == null) {
       throw LanguageError("${ActionType.share.name} requires 'type'");
```

Comment:
careful with cut and paste.

Reply:
The compiler doesn't complain about `return HapticAction(hapticType);`? hapticType can be null but you declare it as non-null.

The hapticType can be an expression, meaning you need to evaluate its value when executing it (not at creation time). See how the rest of the Action work. e.g. scopeManager.eval(hapticType)